redbox-core/redbox/models/chain.py

```diff
@@ -180,11 +180,29 @@ class RedboxQuery(BaseModel):
     ai_settings: AISettings = Field(description="User request AI settings", default_factory=AISettings)
 
 
+class RequestMetadata(TypedDict):
+    input_tokens: int
+    output_tokens: int
+
+
+def metadata_reducer(current: RequestMetadata | None, update: RequestMetadata | None):
+    if current is None:
+        return update
+    if update is None:
+        return current
+    if "input_tokens" in update:
+        input_tokens = current.get("input_tokens", 0) + update["input_tokens"]
```

Comment:
ive been bitten by this before where the RequestMetadata is something like `{"input_tokens": None}` and you end up trying to add None to int or similar. do we need something to coerce None to 0?